plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/RetryDriver.java

```diff
@@ -130,6 +130,11 @@ public <V> V run(String callableName, Callable<V> callable)
                 return callable.call();
             }
             catch (Exception e) {
+                // Immediately stop retry attempts once an interrupt has been received
+                if (e instanceof InterruptedException || Thread.currentThread().isInterrupted()) {
```

Comment:
I think we should recover the `Interrupted` flag while also stopping retries.

Reply:
I was on the fence about whether to do that here. I think since we're re-throwing `InterruptedException` directly, the caller is supposed to be responsible for making that decision- otherwise you can end up setting an "extra" interrupts on the current thread (perhaps inappropriately?).

Reply:
Nevermind, I think I misread. You are re-throwing the `InterruptedException` and not suppressing it. No need to recover the flag.